### ISSUE TITLE
Add support for non-Gmail emails

### ISSUE THREAD

**Issue by naclcaleb** [naclcaleb/EmailDeleter#1]
Currently, due to the way emails are deleted in the code, this probably won't work with anything but Gmail.

Adding support would require email provider detection by getting the `@website.com` from the user's email, then using the appropriate trashing method for that provider.

@OOPS-Studio, @rocketbear27, I thought I'd see if you guys were interested in this. I know Python might not be your preferred language, but still. No worries if you don't want to.

**Comment by rocketbear27** [comment]
@naclcaleb: This sounds like a good project

**Comment by naclcaleb** [comment]
If you want to work on it, that would be great. I most likely won't though, because I'm pretty busy.

I also have a question, for both you and OOPS: What exactly do you like about NodeJS as opposed to other languages?

For example, let's say I want to make a simple website. I could do this with either NodeJS or Python.

With NodeJS, I have to create a `package.json`, figure out the dependencies using `npm`, create a basic server script using the `http` module, and then connect it to the HTML/CSS/JS.

With Python, however, I can easily just `pip install flask`, and then right this simple script:
```
from flask import Flask
app = Flask(__name__)
@app.route("/home")
def home():
    return "<html_source_code>"
if __name__ == '__main__':
    app.run(host='0.0.0.0', port='80')
```
And that's literally it.

And if I wanted to make a more complicated web app with a dashboard and everything, I could just use Django: https://docs.djangoproject.com/en/2.2/intro/tutorial01/

I don't know what rocketbear27 will say, but I assume OOPS will say something about not wanting to use libraries? But in that case my question becomes, "how is NodeJS, HTTP, or Express if you want to use that, not a library?".

**Comment by rocketbear27** [comment]
The first language I learned is JavaScript, I have been extending on my skills and came upon Node.js
I should try creating web applications in Python because it seems equally as easy to set up a brand new application. I haven't tried Python yet, but Node.js has a lot of different useful dependencies for many types of projects. Does Python have a package registry, for example, npm for Node.js?

**Comment by OOPS-Studio** [comment]
Well, I use Node.js because, very similarly to Rocketbear, I'm very fluent in JS and Node.js is pretty similar! ;) And I don't use any libraries when I code with Node... I occasionally use "ws"/"websocket" but other than that... ;)

**Comment by naclcaleb** [comment]
@rocketbear27, yes, Python has `pip`, and there are a lot of packages available for it.

Let's say you need a Python library that makes it easy to make HTTP requests. You could do it the long way, like OOPS might :wink:, or you could `pip install requests` and then `requests.get('url')`.

@OOPS-Studio, but what about Node.js itself? I would consider that a library, and do you understand how that is written? 
As far as I know, NodeJS is written primarily in C and C++, because without that your JS would not be able to do things such as open files, access databases, etc.

NodeJS just seems...artificial to me. IDK, just my perspective.

And also I'm pretty sure you use the default HTTP library every so often...

**Comment by OOPS-Studio** [comment]
1. Yeah... It _kinda_ is... But if you want to look at it that way... Aren't all language libraries! :P Why aren't we all coding in binary?
2. Well, okay... XD
3. Only once! ;)

**Comment by naclcaleb** [comment]
1. True...ish. Except NodeJS is just an extra layer to allow an existing language (JS) to have escalated privileges. Don't get me wrong, I don't hate NodeJS. I use it plenty, especially for frontend due to the plethora of frontend frameworks available (React, React Native, Angular, Vue, lit-html, etc.). I just think it's a bit overrated.

3. How many times have you made a NodeJS project? Literally every single project I've seen with NodeJS as a backend uses the HTTP module.

**Comment by OOPS-Studio** [comment]
1. Meh.
Binary -> Assembly -> C -> C++
Binary -> Assembly -> C -> C++ -> Node.js
It's not too big a leap! ;)
2. *You'll never know...*

**Comment by naclcaleb** [comment]
1. 😆 Yea, but Node.js isn't a language...it's more of a compiler...except it uses other people's compilers...so really it's more like a bunch of C++ code that is abstracted away and called by JS code...so it's the same but...also different. It's just kind of a strange setup. It's like there's two JavaScripts.
Binary -> Assembly -> C -> C++
Binary -> Assembly -> C -> C++ -> JavaScript -> More C++ -> Node.js

(BTW, I don't like calling it `Node.js`, but rather `NodeJS`, since it's not like it's just a JS file or library like we conventionally use the `.js` ending for...)

2. Fine by me...the thing is that JS was never meant to host a server, so it functionally can't, without adding C++ libraries onto it (or other languages, like Python, Ruby, PHP, etc.), so unless you use those languages somewhere (which I doubt), you must be using some sort of HTTP module...

**Comment by naclcaleb** [comment]
Oh, also @OOPS-Studio, I meant to ask whether you knew about the Github Student Developer Pack.

You said you were starting community college soon, so you'd qualify. 

In particular, it gives you a free Heroku Hobby dyno for up to two years, in case you want to actually host a different project. 

But there's plenty of other benefits. You can look at them here: https://education.github.com/pack
